File: Python/721A-japanesecrossword.py
n = int(input())
row = input()
groups = []
count = 0
for i in range(n):
    if row[i] == 'B':
        count += 1

    if row[i] == 'W' or i == n-1:
        if count > 0:
            groups.append(count)
            count = 0
print(len(groups))
for group in groups:
    print(group, end=' ')

File: C++/A/721A-japanesecrossword.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    string row;
    cin >> row;
    vector<int> groups;
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (row[i] == 'B')
            count++;
        if (row[i] == 'W' || i == n - 1)
        {
            if (count > 0)
                groups.push_back(count);
            count = 0;
        }
    }
    cout << groups.size() << '\n';
    for (int i : groups)
        cout << i << " ";
}
